pagecount.c

```diff
@@ -0,0 +1,57 @@
+#include <stdint.h>
+#include <stdio.h>
+#include <time.h>
+
+#define TEST_LOOPS 3333333 // divided by 3 to get rid of mod operation
+
+uint64_t pagecount(uint64_t memory_size, uint64_t page_shift) {
+  memory_size = memory_size >> page_shift;
+  return memory_size;
+}
+
+int main (int argc, char** argv) {
+  clock_t baseline_start, baseline_end, test_start, test_end;
+  uint64_t memory_size, page_size;
+  double clocks_elapsed, time_elapsed;
+  int i, ignore = 0;
+
+  uint64_t msizes[] = {1L << 32, 1L << 40, 1L << 52};
+  uint64_t pshifts[] = {12, 16, 32}; // edited to allow bit shifting instead of division
```

Comment:
Ah, I was thinking of doing something like this too...I couldn't figure out if there was a better way to get the desired shift amount from the divisor. Passing it in directly is probably easiest.

Reply:
Yea I couldn't figure out another way of doing it. xD Sorry for the delayed response